packages/core/versioning/core-version-http-server/src/version_http_toolkit.ts

```diff
@@ -154,12 +112,26 @@ export interface VersionedRouter<Ctx extends RqCtx = RqCtx> {
   options: VersionedRouteRegistrar<'options', Ctx>;
 }
 
+/** @experimental */
+type InValidation<P, Q, B> = RouteValidatorFullConfig<P, Q, B>;
+
+/**
+ * Versioned route validation
+ * @experimental
+ */
+interface FullValidationConfig<P, Q, B, R> {
+  /** Validation to run against route inputs: params, query and body */
+  in?: InValidation<P, Q, B>;
```

Comment:
nit: Similar to https://github.com/elastic/kibana/pull/153011/files#diff-87d84d32aa3e33162d53f5704ac588c2528c456d668e2c34b582a24f1fbe3c93R116, a safer alternative is `input`